File: examples/python/core/parallel/test-concurrency-limits.py
@pytest.mark.durable_execution(
    handler=handler_with_concurrency_limit,
    lambda_function_name="limited_concurrency",
)
def test_concurrency_limit(durable_runner):
    """Test parallel operations with concurrency limit."""
    with durable_runner:
        result = durable_runner.run(input={"items": list(range(20))}, timeout=30)
    
    # All items should be processed
    assert result.status is InvocationStatus.SUCCEEDED
    assert len(result.result["results"]) == 20

File: examples/java/core/parallel/test-concurrency-limits.java
@Test
void testConcurrencyLimit() {
    var runner = LocalDurableTestRunner.create(Map.class, new ControlledConcurrencyHandler());

    List<String> items = IntStream.range(0, 20)
        .mapToObj(String::valueOf)
        .collect(Collectors.toList());

    var result = runner.runUntilComplete(Map.of("items", items));

    // All items should be processed
    assertEquals(ExecutionStatus.SUCCEEDED, result.getStatus());
    var resultMap = result.getResult(Map.class);
    var results = (List<?>) resultMap.get("results");
    assertEquals(20, results.size());
}
